gcloud-java-core/src/main/java/com/google/gcloud/Restorable.java

```diff
@@ -0,0 +1,15 @@
+package com.google.gcloud;
+
+/**
+ * Implementation of this interface can persist their state and restore from it.
+ */
+public interface Restorable<T extends Restorable<T>> {
```

Comment:
We use `Restorable` and `RestorableState` in different classes across the library, do you think that an usage example might help users approaching to the library for the first time? If yes, where should we place it to make it more effective?